app/code/local/Dwyera/Pinpay/Model/Result.php

```diff
@@ -165,6 +165,54 @@ public function getResponseToken()
             throw new Dwyera_Pinpay_Model_ResponseParseException;
         }
     }
+    public function getCustomerToken()
+    {
+        if($this->isResponseParamExists()) {
+            if (isset($this->msgObj->response->token)) {
+                return $this->msgObj->response->token;
+            } else {
+                throw new Dwyera_Pinpay_Model_ResponseParseException;
+            }
+        }
+    }
+
+    public function getPrimaryCardDisplayNumber()
+    {
+        if($this->isResponseParamExists("card")){
+            if (isset($this->msgObj->response->card->display_number)) {
+                return $this->msgObj->response->card->display_number;
+            } else {
+                throw new Dwyera_Pinpay_Model_ResponseParseException;
+            }
+        }
+    }
+
+    public function getCardToken()
+    {
+        if($this->isResponseParamExists("card")){
+            if (isset($this->msgObj->response->card->token)) {
+                return $this->msgObj->response->card->token;
+            } else {
+                throw new Dwyera_Pinpay_Model_ResponseParseException;
+            }
+        }
+
+    }
+
+    public function isResponseParamExists($path = null){
+        if (isset($this->msgObj->response)) {
```

Comment:
@ramya-aligent we had a discussion about moving the exception throwing in this function 
By sending a second parameter which can be `token` or `display_number`